modDbImpl/src/org/aion/db/impl/leveldb/LevelDB.java

```diff
@@ -423,11 +380,31 @@ public void commitBatch() {
     }
 
     @Override
-    public void deleteBatch(Collection<byte[]> keys) {
-        check(keys);
+    public void putBatchInternal(Map<byte[], byte[]> input) {
+        // try-with-resources will automatically close the batch object
+        try (WriteBatch batch = db.createWriteBatch()) {
+            // add put and delete operations to batch
+            for (Map.Entry<byte[], byte[]> e : input.entrySet()) {
+                byte[] key = e.getKey();
+                byte[] value = e.getValue();
```

Comment:
can we just assign the return value into the following method?

Reply:
to avoid the local variable creating increase the heap size and the GC

Reply:
just not sure does 
                byte[] key = e.getKey();
                byte[] value = e.getValue();
                batch.put(key, value);
has any overhead during the runtime?
does this has same memory uausge and performance as just using  batch.put(e.getKey(), e.getValue);